Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Sliders/PinchSlider.cs

```diff
@@ -0,0 +1,198 @@
+        #endregion
+
+
+        #region Unity methods
+        public void Start()
+        {
+            UpdateUI();
+            OnStartRunning.Invoke(new SliderEventData(sliderValue, sliderValue, false));
+        }
+        #endregion
+
+        #region Private Methods
+        private Vector3 GetSliderAxis()
+        {
+            switch (sliderAxis)
+            {
+                case SliderAxis.XAxis:
+                    return Vector3.right;
+                case SliderAxis.YAxis:
+                    return Vector3.up;
+                case SliderAxis.ZAxis:
+                    return Vector3.forward;
+                default:
+                    throw new ArgumentOutOfRangeException("Invalid slider axis");
+            }
+        }
+        private IEnumerator MoveSlider()
+        {
+            while (activePointer != null)
+            {
+                var delta = activePointer.Position - startPointerPosition;
+                var p0ToP1 = SliderEndPosition - SliderStartPosition;
+                var handDelta = Vector3.Dot(p0ToP1.normalized, delta);
+
+                SliderValue = Mathf.Clamp(startSliderValue + handDelta / p0ToP1.magnitude, 0, 1);
+
+                yield return null;
+            }
+        }
+
+        private void UpdateUI()
+        {
+            var p0ToP1 = SliderEndPosition - SliderStartPosition;
+            var newSliderPos = SliderStartPosition + p0ToP1 * sliderValue;
+
+            sliderRoot.transform.position = newSliderPos;
+        }
+
+        #endregion
+
+        #region IMixedRealityFocusHandler
+        public void OnFocusEnter(FocusEventData eventData)
+        {
+            OnHoverEntered.Invoke(new SliderEventData(sliderValue, sliderValue, eventData.Pointer is IMixedRealityNearPointer));
+        }
+
+        public void OnFocusExit(FocusEventData eventData)
+        {
+            OnHoverExited.Invoke(new SliderEventData(sliderValue, sliderValue, eventData.Pointer is IMixedRealityNearPointer));
+        }
+        #endregion
+
+        #region IMixedRealityPointerHandler
+
+        public void OnPointerUp(MixedRealityPointerEventData eventData)
+        {
+            if (OnInteractionEnded != null)
+            {
+                OnInteractionEnded.Invoke(new SliderEventData(sliderValue, sliderValue, activePointer is IMixedRealityPointer));
+            }
+            activePointer = null;
+            if (moveSliderCoroutine != null)
+            {
+                StopCoroutine(moveSliderCoroutine);
+                moveSliderCoroutine = null;
+            }
+        }
+
+        public void OnPointerDown(MixedRealityPointerEventData eventData)
+        {
+            activePointer = eventData.Pointer;
+            startSliderValue = sliderValue;
+            startPointerPosition = activePointer.Position;
+            startSliderPosition = gameObject.transform.position;
+            moveSliderCoroutine = StartCoroutine(MoveSlider());
```

Comment:
Looks like we will be able to use OnPointerDragged here, yeih!

Reply:
Done.